article/adj-difference-matrix.tex

```diff
@@ -0,0 +1,20 @@
+\subsection{Generated adjacency matrix}
+In the following subsection the generated adjacency matrices of LightGCN and price aware recommendation are described.
+
+\subsection{LightGCN}
+For LightGCN the adjacency matrix is the size of $\textrm{number of users} + \textrm{number of items} \times \textrm{number of users} + \textrm{number of items}$.
+Then for every connection between item $i$ and user $u$ 1 is inserted.
+
+\subsection{Price-aware recommendation}
+The adjacency matrix in price-aware recommendation is implemented as a sparse matrix using SpiPy.
+This is generated by creating a row of users and items, and a column with prices and categories.
+The row is generated by mapping the user id $i$ into the row the number of times the user has interacted with an item.
+The column is mapped with the corresponding items that the users has interacted with.
+At this point the length of row and column, would be equal, as both is the length of the total number of interactions.
+When all users and all items has been inserted into the row and column, the prices and categories has to be inserted.
+This is done by adding the index of a item added with total number of users into the row twice.
+For column at the same indexes as the row the category is inserted and the price is inserted next.
+A third array is created with the length of row where ones are inserted.
+Using this array, column and row price aware recommendation create a sparse adjacency matrix by mapping the index $i$ in row and column to the array with ones.
+Within the adjacency matrix, there will be inserted 1 if there is a connection between item $i$ and user $u$, item $i$ and category $c$, and item $i$ and price $p$.
+This results in a adjacency matrix that is of size number of users + number of items + number of categories + number of prices $\times$ number of users + number of items + number of categories + number of prices
```

Comment:
måske være fint at nævne at det her egentlig repræsenterer den heterogene graf. hvor et 1 tal vil sige at der er en edge imellem to nodes.

Den bliver også mirrored senere sådan at row i column j indeholder den samme væredi som row j column i